File: frontend/src/services/crmService.js
// Universelle Imports für Mobile und Web
let AsyncStorage;
try {
  AsyncStorage = require('@react-native-async-storage/async-storage').default;
} catch (e) {
  // AsyncStorage nicht verfügbar (Web-Browser)
  AsyncStorage = null;
}

const API_BASE_URL = 'https://jkb-grounds-production.up.railway.app/api/crm';

const getHeaders = async () => {
  try {
    let token = null;
    
    // PLATTFORM-ERKENNUNG: Web Browser hat Priorität
    if (typeof window !== 'undefined' && typeof localStorage !== 'undefined') {
      // 🌐 WEB BROWSER - ERSTE PRIORITÄT
      console.log('🔍 CRM Service - Web: Lade Token aus localStorage...');
      
      // DEBUG: Zeige localStorage Inhalt (nur im Development)
      if (process.env.NODE_ENV === 'development') {
        console.log('📋 localStorage Inhalt:');
        for (let i = 0; i < localStorage.length; i++) {
          const key = localStorage.key(i);
          const value = localStorage.getItem(key);
          console.log(`  ${key}: ${value}`);
        }
      }
      
      // Suche nach Token in verschiedenen Keys
      token = localStorage.getItem('token') || 
              localStorage.getItem('authToken') ||
              localStorage.getItem('accessToken') ||
              localStorage.getItem('jwt') ||
              localStorage.getItem('access_token') ||
              localStorage.getItem('bearerToken');
      
      console.log('🌐 Web Token gefunden:', token ? 'JA' : 'NEIN');
      
    } else if (AsyncStorage) {
      // 📱 REACT NATIVE MOBILE APP - ZWEITE PRIORITÄT
      console.log('🔍 CRM Service - Mobile: Lade Token aus AsyncStorage...');
      
      try {
        // Suche nach Token in verschiedenen Keys
        token = await AsyncStorage.getItem('token') || 
                await AsyncStorage.getItem('authToken') ||
                await AsyncStorage.getItem('accessToken') ||
                await AsyncStorage.getItem('jwt') ||
                await AsyncStorage.getItem('access_token') ||
                await AsyncStorage.getItem('bearerToken');
        
        console.log('📱 Mobile Token gefunden:', token ? 'JA' : 'NEIN');
        
      } catch (asyncError) {
        console.error('❌ AsyncStorage Fehler:', asyncError);
        token = null;
      }
      
    } else {
      // 🚫 KEINE PLATTFORM ERKANNT
      console.warn('⚠️ Weder localStorage (Web) noch AsyncStorage (Mobile) verfügbar!');
    }
    
    const headers = {
      'Content-Type': 'application/json',
    };
    
    // Token hinzufügen wenn vorhanden
    if (token) {
      headers['Authorization'] = `Bearer ${token}`;
      console.log('🔑 CRM Service - Token hinzugefügt');
      
      // Debug: Token Preview (nur die ersten 30 Zeichen)
      if (process.env.NODE_ENV === 'development') {
        console.log('🔍 Token Preview:', token.substring(0, 30) + '...');
      }
    } else {
      console.warn('⚠️ CRM Service - Kein Token gefunden!');
    }
    
    return headers;
    
  } catch (error) {
    console.error('❌ CRM Service - Fehler beim Token laden:', error);
    return {
      'Content-Type': 'application/json',
    };
  }
};

export const crmService = {
  // Alle bestätigten Nutzer abrufen
  async getUsers() {
    try {
      console.log('🔄 CRM Service - Lade alle Nutzer...');
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/users`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Users Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Users API Error:', response.status, data);
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Laden der Nutzer`);
      }
      
      console.log('✅ CRM Service - Nutzer geladen:', data?.data?.length || 0);
      return data.data || [];
    } catch (error) {
      console.error('CRM Service - getUsers:', error);
      throw error;
    }
  },

  // Ausstehende Registrierungen abrufen
  async getPendingRegistrations() {
    try {
      console.log('🔄 CRM Service - Lade ausstehende Registrierungen...');
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/registrations/pending`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Pending Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Pending API Error:', response.status, data);
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Laden der Registrierungen`);
      }
      
      console.log('✅ CRM Service - Registrierungen geladen:', data?.data?.length || 0);
      return data.data || [];
    } catch (error) {
      console.error('CRM Service - getPendingRegistrations:', error);
      throw error;
    }
  },

  // Registrierung bestätigen
  async approveRegistration(id) {
    try {
      console.log('🔄 CRM Service - Bestätige Registrierung:', id);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/registrations/${id}/approve`, {
        method: 'PUT',
        headers: headers
      });
      
      console.log('📡 Approve Registration Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Approve Registration API Error:', response.status, data);
        
        if (response.status === 404) {
          throw new Error('Registrierung nicht gefunden');
        }
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Bestätigen der Registrierung`);
      }
      
      console.log('✅ CRM Service - Registrierung bestätigt:', data);
      return data;
    } catch (error) {
      console.error('CRM Service - approveRegistration:', error);
      throw error;
    }
  },

  // Registrierung ablehnen
  async rejectRegistration(id) {
    try {
      console.log('🔄 CRM Service - Lehne Registrierung ab:', id);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/registrations/${id}/reject`, {
        method: 'DELETE',
        headers: headers
      });
      
      console.log('📡 Reject Registration Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Reject Registration API Error:', response.status, data);
        
        if (response.status === 404) {
          throw new Error('Registrierung nicht gefunden');
        }
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Ablehnen der Registrierung`);
      }
      
      console.log('✅ CRM Service - Registrierung abgelehnt:', data);
      return data;
    } catch (error) {
      console.error('CRM Service - rejectRegistration:', error);
      throw error;
    }
  },

  // Neuen Nutzer hinzufügen
  async createUser(userData) {
    try {
      console.log('🔄 CRM Service - Erstelle neuen Nutzer:', userData.email);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/users`, {
        method: 'POST',
        headers: headers,
        body: JSON.stringify(userData)
      });
      
      console.log('📡 Create User Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Create User API Error:', response.status, data);
        
        if (response.status === 400) {
          throw new Error('Ungültige Nutzerdaten oder Email bereits registriert');
        }
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Erstellen des Nutzers`);
      }
      
      console.log('✅ CRM Service - Nutzer erstellt:', data);
      return data;
    } catch (error) {
      console.error('CRM Service - createUser:', error);
      throw error;
    }
  },

  // Nutzer bearbeiten
  async updateUser(id, userData) {
    try {
      console.log('🔄 CRM Service - Bearbeite Nutzer:', id);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/users/${id}`, {
        method: 'PUT',
        headers: headers,
        body: JSON.stringify(userData)
      });
      
      console.log('📡 Update User Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Update User API Error:', response.status, data);
        
        if (response.status === 404) {
          throw new Error('Nutzer nicht gefunden');
        }
        if (response.status === 400) {
          throw new Error('Ungültige Nutzerdaten');
        }
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Bearbeiten des Nutzers`);
      }
      
      console.log('✅ CRM Service - Nutzer bearbeitet:', data);
      return data;
    } catch (error) {
      console.error('CRM Service - updateUser:', error);
      throw error;
    }
  },

  // Nutzer löschen
  async deleteUser(id) {
    try {
      console.log('🔄 CRM Service - Lösche Nutzer:', id);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/users/${id}`, {
        method: 'DELETE',
        headers: headers
      });
      
      console.log('📡 Delete User Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Delete User API Error:', response.status, data);
        
        if (response.status === 404) {
          throw new Error('Nutzer nicht gefunden');
        }
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Löschen des Nutzers`);
      }
      
      console.log('✅ CRM Service - Nutzer gelöscht:', data);
      return data;
    } catch (error) {
      console.error('CRM Service - deleteUser:', error);
      throw error;
    }
  },

  // Statistiken abrufen
  async getStats() {
    try {
      console.log('🔄 CRM Service - Lade Statistiken...');
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/stats`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Stats Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Stats API Error:', response.status, data);
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Laden der Statistiken`);
      }
      
      console.log('✅ CRM Service - Statistiken geladen:', data.data);
      return data.data || { total: 0, pending: 0 };
    } catch (error) {
      console.error('CRM Service - getStats:', error);
      throw error;
    }
  },

  // Rollen abrufen
  async getRoles() {
    try {
      console.log('🔄 CRM Service - Lade Rollen...');
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/roles`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Roles Response Status:', response.status);
      
      if (!response.ok) {
        console.log('⚠️ CRM Service - Roles API nicht verfügbar, verwende Fallback');
        // Fallback zu Standard-Rollen
        return [
          { id: 1, name: 'Admin' },
          { id: 2, name: 'Trainer' },
          { id: 3, name: 'Mannschaftsführer' },
          { id: 4, name: 'Mitglied' },
          { id: 5, name: 'Gast' }
        ];
      }
      
      const data = await response.json();
      console.log('✅ CRM Service - Rollen geladen:', data?.data?.length || 0);
      return data.data || [
        { id: 1, name: 'Admin' },
        { id: 2, name: 'Trainer' },
        { id: 3, name: 'Mannschaftsführer' },
        { id: 4, name: 'Mitglied' },
        { id: 5, name: 'Gast' }
      ];
    } catch (error) {
      console.error('CRM Service - getRoles:', error);
      // Fallback zu Standard-Rollen bei Fehler
      console.log('⚠️ CRM Service - Fallback zu Standard-Rollen');
      return [
        { id: 1, name: 'Admin' },
        { id: 2, name: 'Trainer' },
        { id: 3, name: 'Mannschaftsführer' },
        { id: 4, name: 'Mitglied' },
        { id: 5, name: 'Gast' }
      ];
    }
  },

  // Erweiterte Nutzersuche
  async searchUsers(query, filters = {}) {
    try {
      console.log('🔄 CRM Service - Erweiterte Suche:', query, filters);
      
      const headers = await getHeaders();
      
      const searchParams = new URLSearchParams({
        q: query || '',
        ...filters
      });
      
      const response = await fetch(`${API_BASE_URL}/users/search?${searchParams}`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Search Response Status:', response.status);
      
      const data = await response.json();
      
      if (!response.ok) {
        console.error('❌ Search API Error:', response.status, data);
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler bei der Suche`);
      }
      
      console.log('✅ CRM Service - Suche abgeschlossen:', data?.data?.length || 0);
      return data.data || [];
    } catch (error) {
      console.error('CRM Service - searchUsers:', error);
      throw error;
    }
  },

  // Nutzer exportieren
  async exportUsers(format = 'csv') {
    try {
      console.log('🔄 CRM Service - Exportiere Nutzer als:', format);
      
      const headers = await getHeaders();
      
      const response = await fetch(`${API_BASE_URL}/users/export?format=${format}`, { 
        method: 'GET',
        headers: headers
      });
      
      console.log('📡 Export Response Status:', response.status);
      
      if (!response.ok) {
        const data = await response.json();
        console.error('❌ Export API Error:', response.status, data);
        throw new Error(data.detail || `API Fehler ${response.status}: Fehler beim Exportieren`);
      }
      
      const blob = await response.blob();
      console.log('✅ CRM Service - Export erstellt');
      return blob;
    } catch (error) {
      console.error('CRM Service - exportUsers:', error);
      throw error;
    }
  }
};

File: backend/lambda_package/app/services/crm_service.py
from typing import List, Optional, Dict, Any
from supabase import Client
import bcrypt
from datetime import datetime

class CRMService:
    
    @staticmethod
    def get_confirmed_users(supabase: Client, verein_id: str) -> List[Dict[str, Any]]:
        """Alle bestätigten Nutzer für CRM abrufen"""
        try:
            print(f"🔍 DEBUG: CRM sucht bestätigte Nutzer für verein_id: {verein_id}")
            
            # ZEIGE ALLE NUTZER IN DER DB
            all_users = supabase.table('nutzer').select('id, name, email, verein_id, ist_bestaetigt').execute()
            print(f"🔍 DEBUG: ALLE NUTZER in der Datenbank:")
            for user in all_users.data:
                status = "✅ Bestätigt" if user['ist_bestaetigt'] else "❌ Unbestätigt"
                print(f"  - {user['name']} | {user['email']} | Verein-ID: {user['verein_id']} | {status}")
            
            response = supabase.table('nutzer').select('*').eq('ist_bestaetigt', True).eq('verein_id', verein_id).execute()
            print(f"🔍 DEBUG: Gefunden {len(response.data)} bestätigte Nutzer für CRM verein_id {verein_id}")
            
            result = []
            for user in response.data:
                name_parts = (user.get('name') or '').split(' ', 1)
                vorname = name_parts[0] if name_parts else ''
                nachname = name_parts[1] if len(name_parts) > 1 else ''
                
                result.append({
                    'id': user['id'],
                    'vorname': vorname,
                    'nachname': nachname,
                    'email': user.get('email') or '',
                    'geschlecht': user.get('geschlecht') or '',
                    'rolle': CRMService.get_role_name(supabase, user.get('rolle_id')),
                    'rolle_id': user.get('rolle_id'),
                    'passwort': '',
                    'created_at': None
                })
            
            return result
            
        except Exception as e:
            print(f"Fehler beim Abrufen der Nutzer: {e}")
            raise e
    
    @staticmethod
    def get_pending_registrations(supabase: Client, verein_id: str) -> List[Dict[str, Any]]:
        """Ausstehende Registrierungen abrufen"""
        try:
            print(f"🔍 DEBUG: CRM sucht ausstehende Registrierungen für verein_id: {verein_id}")
            
            # ZEIGE ALLE UNBESTÄTIGTEN NUTZER
            all_pending = supabase.table('nutzer').select('id, name, email, verein_id, ist_bestaetigt').eq('ist_bestaetigt', False).execute()
            print(f"🔍 DEBUG: ALLE UNBESTÄTIGTEN NUTZER in der Datenbank:")
            for user in all_pending.data:
                print(f"  - {user['name']} | {user['email']} | Verein-ID: {user['verein_id']}")
            
            response = supabase.table('nutzer').select('*').eq('ist_bestaetigt', False).eq('verein_id', verein_id).execute()
            
            print(f"📊 CRM: {len(response.data)} ausstehende Registrierungen gefunden für CRM verein_id {verein_id}")
            
            result = []
            for user in response.data:
                name_parts = (user.get('name') or '').split(' ', 1)
                vorname = name_parts[0] if name_parts else ''
                nachname = name_parts[1] if len(name_parts) > 1 else ''
                
                result.append({
                    'id': user['id'],
                    'vorname': vorname,
                    'nachname': nachname,
                    'email': user.get('email') or '',
                    'geschlecht': user.get('geschlecht') or '',
                    'rolle': CRMService.get_role_name(supabase, user.get('rolle_id')),
                    'rolle_id': user.get('rolle_id'),
                    'zeittext': 'Unbekannt',
                    'created_at': None
                })
            
            return result
            
        except Exception as e:
            print(f"Fehler beim Abrufen der Registrierungen: {e}")
            raise e
    
    @staticmethod
    def approve_registration(supabase: Client, user_id: str, admin_verein_id: str) -> Dict[str, Any]:  # GEAENDERT: str statt int
        """Registrierung bestätigen"""
        try:
            user_response = supabase.table('nutzer').select('*').eq('id', user_id).eq('verein_id', admin_verein_id).eq('ist_bestaetigt', False).execute()
            
            if not user_response.data:
                raise ValueError("Registrierung nicht gefunden")
            
            user = user_response.data[0]
            
            response = supabase.table('nutzer').update({'ist_bestaetigt': True}).eq('id', user_id).execute()
            
            if not response.data:
                raise ValueError("Fehler beim Bestätigen")
            
            name_parts = (user.get('name') or '').split(' ', 1)
            vorname = name_parts[0] if name_parts else ''
            nachname = name_parts[1] if len(name_parts) > 1 else ''
            
            return {
                'id': user['id'],
                'vorname': vorname,
                'nachname': nachname,
                'email': user.get('email') or '',
                'geschlecht': user.get('geschlecht') or '',
                'rolle': CRMService.get_role_name(supabase, user.get('rolle_id'))
            }
            
        except Exception as e:
            print(f"Fehler beim Bestätigen der Registrierung: {e}")
            raise e
    
    @staticmethod
    def reject_registration(supabase: Client, user_id: str, admin_verein_id: str) -> bool:  # GEAENDERT: str statt int
        """Registrierung ablehnen und löschen"""
        try:
            user_response = supabase.table('nutzer').select('id').eq('id', user_id).eq('verein_id', admin_verein_id).eq('ist_bestaetigt', False).execute()
            
            if not user_response.data:
                raise ValueError("Registrierung nicht gefunden")
            
            response = supabase.table('nutzer').delete().eq('id', user_id).execute()
            
            return True
            
        except Exception as e:
            print(f"Fehler beim Ablehnen der Registrierung: {e}")
            raise e
    
    @staticmethod
    def create_user(supabase: Client, user_data: Dict[str, Any], admin_verein_id: str) -> Dict[str, Any]:
        """Neuen Nutzer direkt erstellen (bestätigt)"""
        try:
            existing_response = supabase.table('nutzer').select('id').eq('email', user_data['email']).execute()
            if existing_response.data:
                raise ValueError("Email bereits registriert")
            
            vollname = f"{user_data['vorname']} {user_data['nachname']}"
            
            hashed_password = bcrypt.hashpw(
                user_data['passwort'].encode('utf-8'),
                bcrypt.gensalt()
            ).decode('utf-8')
            
            rolle_id = CRMService.get_role_id(supabase, user_data['rolle'])
            
            user_insert_data = {
                'name': vollname,
                'email': user_data['email'],
                'passwort': hashed_password,
                'geschlecht': user_data['geschlecht'],
                'rolle_id': rolle_id,
                'verein_id': admin_verein_id,
                'ist_bestaetigt': True
            }
            
            response = supabase.table('nutzer').insert(user_insert_data).execute()
            
            if not response.data:
                raise ValueError("Fehler beim Erstellen des Nutzers")
            
            new_user = response.data[0]
            
            return {
                'id': new_user['id'],
                'vorname': user_data['vorname'],
                'nachname': user_data['nachname'],
                'email': user_data['email'],
                'geschlecht': user_data['geschlecht'],
                'rolle': user_data['rolle'],
                'rolle_id': rolle_id
            }
            
        except Exception as e:
            print(f"Fehler beim Erstellen des Nutzers: {e}")
            raise e
    
    @staticmethod
    def update_user(supabase: Client, user_id: str, user_data: Dict[str, Any], admin_verein_id: str) -> Dict[str, Any]:
        """Nutzer bearbeiten"""
        try:
            print(f"🔧 DEBUG: Update User {user_id}")
            print(f"🔧 DEBUG: User Data: {user_data}")
            print(f"🔧 DEBUG: Admin Verein ID: {admin_verein_id}")
            
            user_response = supabase.table('nutzer').select('*').eq('id', user_id).eq('verein_id', admin_verein_id).execute()
            print(f"🔧 DEBUG: User exists check: {len(user_response.data)} users found")
            
            if not user_response.data:
                print(f"❌ DEBUG: Nutzer nicht gefunden! user_id={user_id}, verein_id={admin_verein_id}")
                raise ValueError("Nutzer nicht gefunden")
            
            existing_user = user_response.data[0]
            vollname = f"{user_data['vorname']} {user_data['nachname']}"
            
            update_data = {
                'name': vollname,
                'email': user_data['email'],
                'geschlecht': user_data['geschlecht']
            }
            
            # ✅ ROLLE HINZUFÜGEN - wenn vorhanden
            neue_rolle_id = existing_user.get('rolle_id')  # Default: alte Rolle
            if user_data.get('rolle') and user_data['rolle'].strip():
                neue_rolle_id = CRMService.get_role_id(supabase, user_data['rolle'])
                update_data['rolle_id'] = neue_rolle_id
                print(f"🔧 DEBUG: Rolle wird aktualisiert: {user_data['rolle']} -> {neue_rolle_id}")
            
            # Passwort nur wenn angegeben
            if user_data.get('passwort') and user_data['passwort'].strip():
                hashed_password = bcrypt.hashpw(
                    user_data['passwort'].encode('utf-8'),
                    bcrypt.gensalt()
                ).decode('utf-8')
                update_data['passwort'] = hashed_password
                print("🔧 DEBUG: Passwort wird aktualisiert")
            
            print(f"🔧 DEBUG: Final update_data: {update_data}")
            
            response = supabase.table('nutzer').update(update_data).eq('id', user_id).execute()
            
            if not response.data:
                raise ValueError("Fehler beim Aktualisieren")
            
            updated_user = response.data[0]
            print(f"🔧 DEBUG: User updated successfully: {updated_user}")
            
            return {
                'id': updated_user['id'],
                'vorname': user_data['vorname'],
                'nachname': user_data['nachname'],
                'email': user_data['email'],
                'geschlecht': user_data['geschlecht'],
                'rolle': user_data.get('rolle', 'Mitglied'),
                'rolle_id': neue_rolle_id  # ✅ GEÄNDERT: Neue rolle_id!
            }
            
        except Exception as e:
            print(f"❌ DEBUG: Exception in update_user: {e}")
            raise e
    
    @staticmethod
    def delete_user(supabase: Client, user_id: str, admin_verein_id: str) -> bool:
        """Nutzer löschen"""
        try:
            user_response = supabase.table('nutzer').select('id').eq('id', user_id).eq('verein_id', admin_verein_id).execute()
            
            if not user_response.data:
                raise ValueError("Nutzer nicht gefunden")
            
            response = supabase.table('nutzer').delete().eq('id', user_id).execute()
            
            return True
            
        except Exception as e:
            print(f"Fehler beim Löschen des Nutzers: {e}")
            raise e
    
    @staticmethod
    def get_stats(supabase: Client, verein_id: str) -> Dict[str, Any]:
        """CRM Statistiken abrufen"""
        try:
            total_response = supabase.table('nutzer').select('id', count='exact').eq('ist_bestaetigt', True).eq('verein_id', verein_id).execute()
            total = total_response.count if total_response.count else 0
            
            pending_response = supabase.table('nutzer').select('id', count='exact').eq('ist_bestaetigt', False).eq('verein_id', verein_id).execute()
            pending = pending_response.count if pending_response.count else 0
            
            return {
                'total': total,
                'pending': pending
            }
            
        except Exception as e:
            print(f"Fehler beim Abrufen der Statistiken: {e}")
            raise e
    
    @staticmethod
    def get_role_name(supabase: Client, rolle_id: Optional[str]) -> str:  # GEAENDERT: Optional[str]
        """Hole echten Rollennamen aus der Datenbank"""
        try:
            if not rolle_id:
                return 'Mitglied'
                
            tables_to_try = ['rollen', 'rolle', 'roles', 'user_roles']
            
            for table_name in tables_to_try:
                try:
                    response = supabase.table(table_name).select('name').eq('id', rolle_id).execute()
                    if response.data:
                        return response.data[0]['name']
                except Exception:
                    continue
            
            return 'Mitglied'
            
        except Exception as e:
            print(f"❌ Fehler beim Abrufen des Rollennamens: {e}")
            return 'Mitglied'
    
    @staticmethod  
    def get_role_id(supabase: Client, role_name: str) -> str:
        """Hole echte Rollen-UUID aus der Datenbank"""
        try:
            print(f"🔍 Suche Rolle: {role_name}")
            
            tables_to_try = ['rollen', 'rolle', 'roles', 'user_roles']
            
            for table_name in tables_to_try:
                try:
                    response = supabase.table(table_name).select('id').eq('name', role_name).execute()
                    if response.data:
                        rolle_id = response.data[0]['id']
                        print(f"✅ Rolle gefunden in {table_name}: {rolle_id}")
                        return rolle_id
                except Exception:
                    continue
            
            # Fallback: Lade erste verfügbare Rolle
            for table_name in tables_to_try:
                try:
                    response = supabase.table(table_name).select('id').limit(1).execute()
                    if response.data:
                        rolle_id = response.data[0]['id']
                        print(f"⚠️ Fallback Rolle aus {table_name}: {rolle_id}")
                        return rolle_id
                except Exception:
                    continue
            
            # Letzter Fallback: Standard-UUID
            print("⚠️ Keine Rollen-Tabelle gefunden, verwende Standard-UUID")
            return '550e8400-e29b-41d4-a716-446655440000'
            
        except Exception as e:
            print(f"❌ Fehler beim Abrufen der Rollen-ID: {e}")
            return '550e8400-e29b-41d4-a716-446655440000'